deploy-service/common/src/main/java/com/pinterest/deployservice/handler/PingHandler.java

```diff
@@ -195,7 +195,9 @@ boolean canDeploy(EnvironBean envBean, String host, AgentBean agentBean) throws
         // TODO use ecache to optimize
         // Make sure we do not exceed allowed number of concurrent active deploying agent
         String envId = envBean.getEnv_id();
-        long parallelThreshold = envBean.getMax_parallel();
+        long totalHosts = environDAO.countTotalCapacity(envBean.getEnv_id(), envBean.getEnv_name(),
```

Comment:
The countTotalCapacity currently contains newly provisioned hosts, missing hosts and active hosts. I modified the logic in #131 to only contain newly provisioned hosts and active hosts. Let me know if this does not make sense to you.